core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockDataMap.java

```diff
@@ -642,7 +643,17 @@ private boolean addBlockBasedOnMinMaxValue(FilterExecuter filterExecuter, byte[]
       byte[][] minValue, String filePath, int blockletId) {
     BitSet bitSet = null;
     if (filterExecuter instanceof ImplicitColumnFilterExecutor) {
-      String uniqueBlockPath = filePath.substring(filePath.lastIndexOf("/Part") + 1);
+      String uniqueBlockPath;
+      String blockName = filePath.substring(filePath.lastIndexOf("/") + 1);
+      if (filePath.contains("/Fact/Part0/Segment_")) {
```

Comment:
Use PR https://github.com/apache/carbondata/pull/2503  and check `CarbonUtil.isStandardCarbonTable` instead of doing contains check

Reply:
`CarbonTable` is available in BlockletDataMapModel, so you can do the check.